packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -1100,27 +1100,17 @@ class TextureAndroidViewController extends AndroidViewController {
     );
   }
 
-  /// Creates the Android View.
-  ///
-  /// This should not be called before [AndroidViewController.setSize].
-  ///
-  /// Throws an [AssertionError] if view was already disposed.
-  @override
-  Future<void> create() async {
-    if (_initialSize != null)
-      return super.create();
-  }
-
   @override
   Future<void> _sendCreateMessage() async {
-    assert(_initialSize != null, 'trying to create $TextureAndroidViewController without setting an initial size.');
-    assert(!_initialSize!.isEmpty, 'trying to create $TextureAndroidViewController without setting a valid size.');
+    // The initial size is set in the first call to setSize.
+    final Size size = await _initialSize.future;
```

Comment:
It is not obvious the first call to `setSize` is guaranteed to be called before `_sendCreateMessage`. For example, just by looking at this code, I would think the future can be waited forever and be timed out.
Is there a way to refactor the code so that it is more obvious and  guaranteed.

Reply:
added some clarification in https://github.com/flutter/flutter/pull/101399/commits/93eb4d2b3954d5470df7ccff16c3dd1f5b6a2c45

Reply:
I added some comments. https://github.com/flutter/flutter/commit/2e182852ba50bb7838e216ce037106b130a53126

The contract of the controller is that a size must be given by calling `setInitialSize`, which can be called before or after `create()` is called.

Reply:
Done